plugins/org.python.pydev.parser/src/org/python/pydev/parser/prettyprinterv2/MakeAstValidForPrettyPrintingVisitor.java

```diff
@@ -176,9 +176,14 @@ public static void makeValid(SimpleNode node) throws Exception {
     public Object visitAssign(Assign node) throws Exception {
         fixNode(node);
 
-        for (int i = 0; i < node.targets.length; i++) {
-            exprType target = node.targets[i];
-            target.accept(this);
+        if (node.targets.length > 0) {
+            for (int i = 0; i < node.targets.length; i++) {
+                exprType target = node.targets[i];
+                target.accept(this);
+                nextCol();
+            }
+        } else {
+            node.type.accept(this);
```

Comment:
I think that visiting `node.type` should be done always, no?

i.e. don't put it in the `else` as we want to visit it even if `node.targets.length > 0` and check if `node.type != null` before visiting.

Reply:
Is it for the `var: int = 10` look alike cases? Because when the variable is type declared, the assignment targets comes empty, and all the assigment infos are stored in node.type

Reply:
Yes.

Reply:
But what about the cases assignment is not typed? The refactor should visit node.type and type the assignment or just visit it but leaving as it is?

Reply:
> But what about the cases assignment is not typed? The refactor should visit node.type and type the assignment or just visit it but leaving as it is?

I just saw that there is no need to having that complication. node.type is null when the assignment is not typed, so no problems at all to always visit it ;)